main/src/main/scala/sbt/Main.scala

```diff
@@ -603,7 +603,10 @@ object BuiltinCommands {
         LineRange(0, 0)
       )(cl)
       val setResult =
-        if (all) SettingCompletions.setAll(extracted, settings)
+        if (all) {
+          if (arg == null || arg.trim.isEmpty) s.log.info(s"$arg will be ignored in set All")
```

Comment:
Note that `settings` that is passed in is created using `arg`, so to say arg will be ignored, I don't think is correct. The problem is that when the `set every` takes a scoped  key, the _scoping_ gets ignored.

For example:

```scala
set every version := "1.0.0-M1"
```

this is ok.

```scala
set every Compile / doc / sources := Nil
```

this does not work.

Reply:
As part of the issue, it was asked to add a log statement in case scope is provided in set every. 
Should I make change to the semantics to use the scope?
@eed3si9n

Reply:
What Toni asked in the issue was this:

> Possible approaches: "set every" should either stop if a scope is supplied in the command line, or at least issue a warning.

Could you run sbt with your modification and show what output it currently prints out for `set every version ...` and `set every Compile / doc / sources` please?